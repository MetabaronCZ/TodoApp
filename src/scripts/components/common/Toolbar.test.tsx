import React from 'react';

import { render } from '@testing-library/react';
import { describe, expect, it } from '@jest/globals';

import { Toolbar } from 'components/common/Toolbar';
import { withMockedProviders } from 'test/component';

describe('components/common/Toolbar', () => {
  it('should render correctly', () => {
    const tree = render(
      withMockedProviders(<Toolbar items={[<div key={0}>CONTENT</div>]} />),
    );
    expect(tree.container).toMatchSnapshot();
    expect(tree.getByText('CONTENT')).toBeInTheDocument();
  });

  it('should render children', () => {
    const testData: React.ReactNode[] = [
      <div key={0}>Item 1</div>,
      <div key={1}>Item 2</div>,
      'filler',
      <div key={2}>Item 3</div>,
      null,
    ];
    const { container } = render(
      withMockedProviders(<Toolbar items={testData} />),
    );
    expect(container).toMatchSnapshot();

    const items = Array.from(container.firstChild?.childNodes ?? []);
    expect(items.length).toEqual(4);

    expect(items.map((item) => item.textContent)).toEqual([
      'Item 1',
      'Item 2',
      '',
      'Item 3',
    ]);
  });
});
