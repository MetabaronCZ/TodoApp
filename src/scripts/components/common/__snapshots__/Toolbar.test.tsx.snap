// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/common/Toolbar > should render children 1`] = `
.c0 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
  height: calc(30px + 1px);
  border-bottom: 1px solid #c0c0c0;
  background: #f0f0f0;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div>
  <div
    class="c0"
  >
    <div>
      <div>
        Item 1
      </div>
    </div>
    <div>
      <div>
        Item 2
      </div>
    </div>
    <div
      class="c1"
    />
    <div>
      <div>
        Item 3
      </div>
    </div>
  </div>
</div>
`;

exports[`components/common/Toolbar > should render correctly 1`] = `
.c0 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
  height: calc(30px + 1px);
  border-bottom: 1px solid #c0c0c0;
  background: #f0f0f0;
}

<div>
  <div
    class="c0"
  >
    <div>
      <div>
        CONTENT
      </div>
    </div>
  </div>
</div>
`;
